torch/hub.py

```diff
@@ -120,28 +121,57 @@ def _read_url(url):
 
 
 def _validate_not_a_forked_repo(repo_owner, repo_name, branch):
+    # Here we try to make sure that the branch/commit hash effectively belongs
+    # to the repo_owner/repo_name repo.
+    # This is important because GitHub is weird, and the download URL
```

Comment:
> weird

Although this gave me a moment of chuckle, I am afraid this might be taken as offensive. I imagine there was some technical reasons behind this for the sake of scalability or something.